frontend/packages/dev-console/src/components/import/devfile/DevfileImportForm.tsx

```diff
@@ -23,8 +29,59 @@ const DevfileImportForm: React.FC<FormikProps<FormikValues> & DevfileImportFormP
 }) => {
   const { t } = useTranslation();
   const [, devfileParseError] = useDevfileServer(values, setFieldValue);
+  const searchParams = new URLSearchParams(window.location.search);
+  const gitRepoUrl = searchParams.get('gitRepo');
+  const devfileName = searchParams.get('devfileName');
+  const formType = searchParams.get('formType');
+  const [devfileSamples, setDevfileSamples] = React.useState<DevfileSample[]>();
+  const selectedSample = React.useMemo(
+    () => devfileSamples?.find((sample) => sample.name === devfileName),
+    [devfileSamples, devfileName],
+  );
   useDevfileDirectoryWatcher(values, setFieldValue);
 
+  React.useEffect(() => {
+    if (gitRepoUrl) {
+      const gitType = detectGitType(gitRepoUrl);
+      const gitService = getGitService({ url: gitRepoUrl }, gitType as GitProvider);
+      setFieldValue('devfile.devfileSourceUrl', gitRepoUrl);
+      gitService
+        .getDevfileContent()
+        .then((contents) => setFieldValue('devfile.devfileContent', contents))
+        .catch(() => {
+          setFieldValue('devfile.devfileContent', null);
+          setFieldValue('devfile.devfileHasError', true);
+        });
+
+      if (formType === 'sample') {
+        setFieldValue('name', createComponentName(devfileName));
+        setFieldValue('git.url', gitRepoUrl);
+        setFieldValue('git.type', gitType);
+        const defaultAppGroup = 'sample-app';
```

Comment:
It would be good to keep this in a const file because the same is being used in other sample form as well.